src/napari_epitools/_widget.py

```diff
@@ -240,3 +247,173 @@ def run() -> tuple[npt.NDArray[np.float64], npt.NDArray[np.int64]]:
+        options={"tooltip": run_tooltip},
+    )
+
+    export_tooltip = (
+        "Export the cell statistics for the selected Image and Labels to a CSV file"
+    )
+    export = magicgui.widgets.create_widget(
+        name="export",
+        label="Export statistics",
+        widget_type="PushButton",
+        options={"tooltip": export_tooltip},
+    )
+
+    return [image, labels, run, export]
+
+
+def _update_cell_statistics(
+    layers: list[napari.layers.Layer],
+    frame: int,
+) -> None:
+    """Update Labels cell_statistics for current frame"""
+
+    for layer in layers:
+        try:
+            layer.features = layer.metadata["cell_statistics"][frame]
+        except KeyError:
+            pass
+        except IndexError:
+            pass
+
+
+def run_cell_statistics(
+    image: napari.layers.Image,
+    labels: napari.layers.Labels,
+) -> None:
+    """Calculate cell statistics for all frames in the selected Image and Labels"""
+
+    cell_statistics, graphs = calculate_cell_statistics(
+        image=image.data,
+        labels=labels.data,
+    )
+
+    # We will use these to update the cell stats at each frame
+    labels.metadata["cell_statistics"] = cell_statistics
+    labels.metadata["graphs"] = graphs
+
+    # confirm calculation finished
+    message = f"'Finished calculating cell statistics for '{labels.name}'"
+    napari.utils.notifications.show_info(message)
+
+    # Set cell stats for the current frame
+    viewer = napari.current_viewer()
+    current_frame = viewer.dims.current_step[0]
+    try:
+        labels.features = cell_statistics[current_frame]
+    except IndexError:
+        pass
+
+
+def export_cell_statistics(
+    labels: napari.layers.Labels,
+) -> None:
+    """Get filename for exporting Labels cell statitics to CSV"""
+
+    app = magicgui.application.use_app()
+    show_file_dialog = app.get_obj("show_file_dialog")
+    filename = show_file_dialog(
+        mode=FileDialogMode.OPTIONAL_FILE,
+        caption="Specify file to save Epitools cell statistics",
+        start_path=None,
+        filter="*.csv",
+    )
+
+    if filename is None:
+        return
+
+    _cell_statistics_to_csv(
+        filename=filename,
+        labels=labels,
+    )
+
+
+def _cell_statistics_to_csv(
+    filename: os.PathLike,
+    labels: napari.layers.Labels,
+) -> None:
+    """Write cell statistics for all frames in the selected Labels to CSV"""
+
+    try:
+        cell_statistics = labels.metadata["cell_statistics"]
+    except KeyError:
+        message = f"'{labels.name}' has no cell statistics to export"
+        napari.utils.notifications.show_error(message)
+        return
+
+    df = pd.concat(
+        [pd.DataFrame.from_dict(stats).set_index("index") for stats in cell_statistics],
+        keys=list(range(len(cell_statistics))),
+    )
+    df.to_csv(filename, index_label=["frame", "label"])
+
+    # confirm export
+    message = f"'{labels.name}' cell statistics written to {filename}"
```

Comment:
Are the first quotes needed? If so we should maybe do both?
```suggestion
    message = f"'{labels.name}' cell statistics written to '{filename}'"
```